File: hw/vizcaino/u3/hw18OOPFarm/hw18Python/model/cow.py
from model.farm_animal import FarmAnimal
from model.meat_animal import IMeatAnimal


class Cow(FarmAnimal, IMeatAnimal):
    def __init__(self, id, breed, born_on, weight, health_status, is_producing_milk, daily_milk_yield):
        super().__init__(id, breed, born_on, weight, health_status)
        self.is_producing_milk = is_producing_milk
        self.daily_milk_yield = daily_milk_yield

    def milk(self):
        return self.daily_milk_yield if self.is_producing_milk else 0

    def cut(self):
        return {"type": "Beef", "yield": 0.65}

    def send_to_slaughter_house(self, slaughter_house):
        slaughter_house.receive(self)

    def to_dict(self):
        return self.__dict__

File: hw/vizcaino/u3/hw18OOPFarm/hw18Java/OOPConceptsFarm/src/main/java/ec/edu/espe/oopconceptsfarm/model/Cow.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.oopconceptsfarm.model;

/**
 *
 * @author Adrian Vizcaino <The-Softwarriors at ESPE>
 */
import java.util.ArrayList;
import java.util.Date;

public class Cow extends FarmAnimal implements IMeatAnimal {

    private boolean isProducingMilk;
    private float dailyMilkYield;
    private float totalMilkProduced;

    public Cow(int id,
            String breed,
            Date bornOn,
            float weight,
            String healthStatus,
            boolean isProducingMilk,
            float dailyMilkYield) {

        super(id, breed, bornOn, weight, healthStatus);

        this.isProducingMilk = isProducingMilk;
        this.dailyMilkYield = dailyMilkYield;
        this.totalMilkProduced = 0;
    }

    public void milk() {

        if (isProducingMilk) {
            System.out.println(
                    "Milk produced: "
                    + dailyMilkYield + " liters");
        }
    }
    public void addMilkProduction(float liters){
    totalMilkProduced += liters;
}

public float getTotalMilkProduced(){
    return totalMilkProduced;
}

    @Override
    public ArrayList<Cut> cut() {

        ArrayList<Cut> cuts = new ArrayList<>();

        cuts.add(
                new Cut(
                        1,
                        "Beef",
                        "Standard Beef Process",
                        weight * 0.65f));

        return cuts;
    }

    @Override
    public void sendToSlaughterHouse(
            SlaughterHouse slaughterHouse) {

        System.out.println(
                "Cow sent to "
                + slaughterHouse.getName());
    }
}
